--- lispd_nat_lib.c	2012-04-19 19:01:40.000000000 +0200
+++ lispd_nat_lib_mod.c	2012-04-20 13:21:16.577522588 +0200
@@ -61,6 +61,12 @@
 #include "linux/netlink.h"
 #include "lispd_external.h"
 
+//modified by arnatal
+/* Temporal. For NAT traversal testing implementation purpose. 
+ * To be removed when it's done
+ */ 
+#define RTR_TEST_RLOC	"192.168.56.102"
+#define RTR_TEST_RLOC_AFI	AF_INET
 
 /*
  * void parameter function just to avoid to write the default values of
@@ -86,6 +92,10 @@
     } PATRICIA_WALK_END;
 
     syslog(LOG_DAEMON, "Sending Info-Request packet");
+    
+    //Until Map Server support Info Request we hardcoded a Dummy Map-Server/RTR 
+
+    lisp_addr_t dummy_server = inet_lisp_addr(RTR_TEST_RLOC, RTR_TEST_RLOC_AFI);
 
     build_and_send_info_request(build_nonce((unsigned int) time(NULL)),
                                 map_servers->key_type,
@@ -95,7 +105,7 @@
                                 &eid_address,
                                 &(ctrl_iface->AF4_locators->head->db_entry->locator),
                                 LISP_CONTROL_PORT,
-                                map_servers->address,
+                                &dummy_server, //map_servers->address,
                                 LISP_CONTROL_PORT);
 
 
@@ -747,7 +757,7 @@
  *
  */
 
-static uint8_t *build_data_encap_pkt(orig_pkt,orig_pkt_len,addr_from,addr_dest,
+uint8_t *build_data_encap_pkt(orig_pkt,orig_pkt_len,addr_from,addr_dest,
                               port_from,port_dest,data_encap_pkt_len)
 
 uint8_t * orig_pkt;
